WORKSPACE.bazel

```diff
@@ -22,13 +22,13 @@ load("@build_bazel_rules_nodejs//:index.bzl", "node_repositories", "yarn_install
 # Setup the Node.js toolchain for the architectures we want to support
 node_repositories(
   node_repositories = {
-    "16.20.1-darwin_amd64": ("node-v16.20.1-darwin-x64.tar.gz", "node-v16.20.1-darwin-x64", "d1f9c2a7c3a0fe09860f701af5fb8ff9ac72d72faa7ebabfeb5794503e79f955"),
-    "16.20.1-darwin_arm64": ("node-v16.20.1-darwin-arm64.tar.gz", "node-v16.20.1-darwin-arm64", "5f6b31c5a75567d382ba67220f3d7a2d9bb0c03d8af9307cd35a9cb32a6fde9d"),
-    "16.20.1-linux_arm64": ("node-v16.20.1-linux-arm64.tar.xz", "node-v16.20.1-linux-arm64", "7fce19f3d1c2952599a0b47f9f5d8f497265ad577f37f256a8c6a03be6353234"),
-    "16.20.1-linux_amd64": ("node-v16.20.1-linux-x64.tar.xz", "node-v16.20.1-linux-x64", "b6c60e1e106ad7d8881e83945a5208c1b1d1b63e6901c04b9dafa607aff3a154"),
-    "16.20.1-windows_amd64": ("node-v16.20.1-win-x64.zip", "node-v16.20.1-win-x64", "2a7fde996c57a969f0498742f99385a520eb14aac864e0eff9c32e3f3633ff0a"),
+    "18.16.1-darwin_amd64": ("node-v18.16.1-darwin-x64.tar.gz", "node-v18.16.1-darwin-x64", "3040210287a0b8d05af49f57de191afa783e497abbb10c340bae9158cb51fdd4"),
```

Comment:
The hashes we give here belong to the official node.js distros, which I assume is fine, because these are binaries related to development, and not the ones we bundle with, right?

I wonder if we should change this bit to also load these files from our mirror, and use the custom-built variants on linux (for less hidden differences) - also using the proxy-cache mirror from line:32 might do away with a few of the Bazel cache warming job failures.

@elastic/kibana-operations -  What do you think?

Reply:
I think it's a good idea, it could help us catch issues earlier.

Do we know if native binaries on the node distribution we're building are compatible with official distributions?  Will a developer with a global, official installation of node and a bazel sandbox build with our glibc compatible version of something like `node-sass` work?

Reply:
I don't think we need to worry about this tbh. Bazel should be short lived and its for development only. In addition to that Bazel has local caches so after fetching once it won't download the binaries again from the nodejs official mirror. Having that said I'm not opposed to change those lines to use our proxy cache links but I think there was something in the version of the bazel js rules we are in that was preventing us from doing that (not completely sure about this at the moment) and, again, as Bazel will be short lived I think we can avoid touching this for now 🙂

Reply:
it seems it's a good catch, @jbudz - I got kibana startup issues, then I tried requiring manually:

```bash
[root@a63630b977b7 kibana]# ./node/bin/node 
Welcome to Node.js v18.16.1.
Type ".help" for more information.
> require('re2')
Uncaught:
Error: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /opt/kibana/node_modules/re2/build/Release/re2.node)
```

When using a prebuilt re2 artifact (from earlier versions of the custom build pipeline), it can be imported, kibana works.

I tried the other way around, it seems the glibc217 built artifacts are going to be usable by other linux platforms too.

Reply:
If I'm reading this correctly, does it means we will have to build and push node-sass and re2 native modules as part of our custom node js build pipeline to make sure we are always building those using glibc2.17? Otherwise they will fail in the machines consuming our custom node images. Is that what you were trying to mention here Alex? If thats the case we will probably have to recover your previous work for node-sass and re2 in the custom nodejs pipeline.

Reply:
Yes, I think. This is my impression from the tests. We'll need the libraries compiled using the custom node builds. 

However, they're not exactly bound to the specific node.js version we're compiling them with, but the ones we have now will probably work for the whole Node 18.x.x family - so long as the modules versions (and apparently other native lib versions) don't change. We could probably live with the few pre-compiled binaries we have produced earlier, but for the long game, it's probably best to build them periodically.